### ISSUE TITLE
Carriers that self destruct will not deploy their fighters

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#4130]
**Describe the bug**
tl;dr I think that when Sestor carriers self destruct with fighters undeployed, the fighters disappear from the game until the player lands/reloads the save/reloads the game.

Currently, destroying a carrier will cause it to deploy its fighters, if there are any. Capturing a carrier with undeployed fighters will cause the fighters to be captured for your. But I believe that there is currently an unhandled situation for when the carrier self destructs. The reason I think this is that while doing the mission to eliminate the Sestor fleets assaulting human space, I attempted to capture any ship I disabled. As expected, many self destructed, including a number of carriers. I then double checked the system I was in before leaving each time. Upon returning to Danforth, he told me there were still Sestor ships to be killed. I checked my save file for where they were, and noticed that all the remaining ships were 1) fighters and drones that 2) came in numbers large enough to fill a carrier and 3) were fully healed. The reason I mention the fully healed part is because Sestor fighters are unable to repair themselves, so fully healed fighters means that one would have expected them to have been docked on a carrier that is able to repair them when the player has landed, ensuring that they have full health. So where ever these fighters were, they were considered to be docked on a carrier.

**To Reproduce**
1. Disable a Sestor carrier from a mission where it must be killed while it still has drones undeployed. 
2. Board it and cause it to self destruct.
3. Land and reload the game (although simply landing and leaving the planet or reloading the save may work too, I wasn't able to pin down what exactly needs to occur).
4. Return to the system that the Sestor carrier self destructed in and notice that there are now fully healed fighters present.

**Expected behavior**
When the carrier self destructs, the fighters should depart from it just as they do when the carrier is normally made to explode through damage.

**System:**
- OS: Windows 10 Home, Version 10.0.17134 Build 17134
 - Game Source: Self-compiled
 - Version: Most up to date as of this issue (https://github.com/endless-sky/endless-sky/commit/b30f509d4f2758c05b19139794c31c801249a9c6)

**Additional Info**
I'm not entirely certain if this is what is causing the disappearance of the fighters that then reappear, it's simply a guess based off of when I know fighters are handled when something happens to the carrier. Whatever the case may be, there is currently something up with fighters disappearing from the game then reappearing later that is very noticeable during the Sestor Remnant mission.

**Comment by VonZorch** [comment]
I would expect the drones/fighters to be destroyed with the self destructing ship.

**Comment by tehhowch** [comment]
I've added 2 commits addressing this - 7d1301faedc to destroy any ships still docked when the carrier explodes, and a769c110cdbf to give any docked ships an extra shot at escaping before it explodes.

This (very) long-lived bug is mostly noticeable in self-destruct scenarios because the ship explodes very soon after the action, while traditional explosions ramp up slowly (giving docked ships time to depart). Additionally, it isn't noticeable by the player until the next launch, when the NPCs are re-added to the ships that are in flight, which means it was solely a "problem" for mission NPCs that are carriers that have to be killed or disabled.
